pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/GrowableArrayBlockingQueue.java

```diff
@@ -401,6 +413,24 @@ public String toString() {
         return sb.toString();
     }
 
+    /**
+     * Make the queue not accept new items. if there are still new data trying to enter the queue, it will be handed
+     * by {@param itemAfterTerminatedHandler}.
+     */
+    public void terminate(@Nullable Consumer<T> itemAfterTerminatedHandler) {
+        terminated = true;
+        if (itemAfterTerminatedHandler != null) {
+            this.itemAfterTerminatedHandler = itemAfterTerminatedHandler;
+        }
+        // After wait for the in-flight item enqueue, it means the operation of terminate is finished.
+        long stamp = tailLock.writeLock();
+        tailLock.unlockWrite(stamp);
```

Comment:
```suggestion
        // After wait for the in-flight item enqueue, it means the operation of terminate is finished.
        long stamp = tailLock.writeLock();
        terminated = true;
        if (itemAfterTerminatedHandler != null) {
            this.itemAfterTerminatedHandler = itemAfterTerminatedHandler;
        }
        tailLock.unlockWrite(stamp);
```

Is it better to understand?